File: Dot examples/ASP Book Examples/Book Examples/Chapter29/VB/App_Code/FullyRenderedControl.vb
Imports System.Web.UI

Namespace myControls
    Public Class FullyRenderedControl
        Inherits Control

        Private _Text As String

        Public Property Text() As String
            Get
                Return _Text
            End Get
            Set(ByVal Value As String)
                _Text = value
            End Set
        End Property

        Protected Overrides Sub Render(ByVal writer As HtmlTextWriter)
            writer.Write(_Text)
        End Sub

    End Class
End Namespace

File: Dot examples/ASP Book Examples/Book Examples/Chapter29/CS/App_Code/FullyRenderedControl.cs
using System.Web.UI;

namespace myControls
{
    public class FullyRenderedControl : Control
    {
        private string _Text;

        public string Text
        {
            get { return _Text; }
            set { _Text = value; }
        }

        protected override void Render(HtmlTextWriter writer)
        {
            writer.Write(_Text);
        }
    }
}